### PR TITLE
Add reproducer for "typed property must not be accessed before initialization..." bug

### PR DESCRIPTION
**Pull request by kbond**
This is a reproducer for #10336.

_The test passes on 2.13.x_

### CODE REVIEW

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH10336Test.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Tests\Models\GH10336\GH10336Entity;
+use Doctrine\Tests\Models\GH10336\GH10336Relation;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @requires PHP 8.0
```
Comment (simonberger) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH10336Test.php]: PHP 8 is not needed for the test.

**Comment by nicolas-grekas** [comment]
Can you please add back the lines mentioned in https://github.com/doctrine/orm/issues/10336#issuecomment-1368578980 to see the effect on tests?

**Comment by kbond** [comment]
Re-adding the lines fixed this specific test but breaks `MappedSuperclassTest`.

**Comment by nicolas-grekas** [comment]
Great thanks. And the break appears only with the new proxies. Looks like they brought some behavior change that we need to understand.

**Comment by simonberger** [comment]
There is another problem in the `ProxyFactory::generateSkippedProperties` ignoring private properties of parent classes which the id property of the test classes is. I can implement a solution or at least prototype later today.

**Comment by nicolas-grekas** [comment]
Fixed in #10385. This PR can be closed. Thanks for the reproducer!

**Comment by kbond** [comment]
Great! Thanks @nicolas-grekas!

**Pull request closed without merging**